pkg/dependencies/factory.go

```diff
@@ -241,6 +242,37 @@ func (f *Factory) WithExecutableImage() *Factory {
 	return f
 }
 
+// WithCustomExecutableImage sets the right cli tools image for the executable builder based on bundlesOverride.
+// If bundleOverride is not set then WithExecutableImage is called.
+// Otherwise, the bundlesOverride is used and the executable image is set to the first VersionsBundle.
+func (f *Factory) WithCustomExecutableImage(bundlesOverride string) *Factory {
+	f.WithManifestReader()
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
```

Comment:
this can be problematic if the executable builder gets built before this does

What I would suggest is to create a `WithCustomBundles` method that stores the bundles URL in `factory.config.bundlesURL`
And then update the logic in `WithExecutableImage` to handle that field if present (basically using the code you have here already)

This explanation might be a bit confusing now I realize, just ping me and let's chat about this offline if it's not clear :)

Reply:
https://github.com/junshun/eks-anywhere/pull/1